middleware/packet-forward-middleware/router/keeper/keeper.go

```diff
@@ -218,7 +219,7 @@ func (k *Keeper) ForwardTransferPacket(
 
 	// pay fees
 	if feeAmount.IsPositive() {
-		hostAccAddr, err := sdk.AccAddressFromBech32(receiver)
+		hostAccAddr, err := sdk.AccAddressFromBech32(feePayer)
```

Comment:
the `intermediateSender` should be the fee payer here, as that is where the funds are received, and will be forwarded out of.